### ISSUE TITLE
🚀  Support custom hyperlink hash formats

### ISSUE THREAD

**Issue by HarrisonMc555** [dandavison/delta#613]
First of all, thanks for the great tool!

I think it's super cool that delta supports terminal hyperlinks. There's a lot of great support and documentation for file links, including a `--hyperlinks-file-link-format` flag that lets you customize how the file links are created.

There is also support for creating links for Git hashes. However, it appears that this is hard-coded to support GitHub repositories and nothing else.

Would you be open to adding a `--hyperlinks-hash-link-format` (or something like that)? I imagine it would be similar and simply have a `{hash}` placeholder.

If we wanted to go further, we could support the `{hash}` placeholder in the file link, which would allow users to open file links in their browser (or something) if they wanted. Just a thought :smiley:

**Comment by dandavison** [comment]
Hi @HarrisonMc555, thanks! Currently what delta does is look in the git repo config for `remote.origin.url` and parse that. And as you say, I only implemented parsing of Github URLs in my first pass and haven't yet gone back to add more. What do you think -- we could add parsing of URLs from other common git hosting sites pretty easily, and then it would be automatic, without the user having to construct a `hash-link-format` argument. Of course, the `hash-link-format` is ultimately more flexible, and we can do both. Ref https://github.com/dandavison/delta/issues/257#issuecomment-663019821

**Comment by dandavison** [comment]
Looking into it more, I see that if the `remote.origin.url` is something like `git@invent.kde.org:utilities/konsole.git` then there doesn't seem much way for delta to know that that is a custom GitLab site (which I believe is the case) and hence that a commit should be formatted like `https://invent.kde.org/utilities/konsole/-/commit/f8e06d55a2d2a7e322cc980052aa86d67d3aae17`, so it does seem like we need the general `hash-link-format` argument, and perhaps also some auto-parsing of well-known formats.

**Comment by HarrisonMc555** [comment]
@dandavison I agree with your conclusions. Parsing common formats (GitHub, GitLab, Bitbucket?) still seems like a nice default, but providing a `hash-link-format` would be necessary for any custom site. For example, my work uses a self-hosted Bitbucket server, which you wouldn't know just from the URL.

If you're open to it I'm willing to look into creating a pull request! I'm more than willing to take any pointers for getting started, though :smiley:

**Comment by dandavison** [comment]
> If you're open to it I'm willing to look into creating a pull request!

Ah, I should have held off :) I actually already made the change (https://github.com/dandavison/delta/pull/614) but if you are interested in helping with some code contributions for delta that would be fantastic and I can always suggest tickets of different levels of effort (feel free to chat here or you can email me at dandavison7@gmail.com). Also understand if it was this particular feature that was providing sufficient motivation :)

**Comment by dandavison** [comment]
OK, this has been added, thanks for the suggestion and sorry for stealing your PR! So for example, on master, I am able to do

```
git -c delta.hyperlinks-commit-link-format='https://invent.kde.org/utilities/konsole/-/commit/{commit}' log
```

and the commit is formatted as a hyperlink to https://invent.kde.org/utilities/konsole/-/commit/1cda384ec34e70ce052f8d177a8afcb665964663.

**Comment by HarrisonMc555** [comment]
No worries! That was a quick turnaround :smiley: thanks for the new feature!

**Comment by dandavison** [comment]
`hyperlinks-commit-link-format` is released now (delta 0.8.0)
